### DISCUSSION TITLE
NSGA-II hyperparams

### DISCUSSION THREAD

**Discussion by taikichi-pc** [optuna/optuna#4528] [Q&A]
Thank you for developing this great library.
Can I ask you a question?

## Question
I do experiment about efficiency of optimization with some test function like below.
[ZDT1~3](https://pymoo.org/problems/multi/zdt.html)
The number of object is 2.

I am using NSGA-II by optuna, but the optimization is not good. 
It's the same or worse than random suggestion.

Could this situation happen?
Or Do I have to set(change) some hyperparams?

### Version
python=3.9
optuna=2.10.1

### Code
I'm using `ask`.
```
def add_trial_to_study(study, study_search_space, problem, new_x = None):
    new_params = {}
    if new_x is None:
        trial = study.ask()
        new_x = []
        # pick up new x
        for i, k in enumerate(study_search_space.keys()):
            new_x.append(trial.suggest_float(k, problem.bounds[0][i], problem.bounds[1][i]))
        new_x = torch.tensor(new_x, device=device)
    for i, k in enumerate(study_search_space.keys()):
        new_params[k] = new_x[i] 
    trial = optuna.trial.create_trial(
        params = new_params,
        distributions=study_search_space,
        values=problem(new_x)
    )
    study.add_trial(trial)
    return study, new_x.reshape(1,-1), problem(new_x).reshape(1,-1)

def initialize_nsgaii(train_x, problem, seed):
    study = optuna.create_study(
        directions=['maximize'] * OBJECT_DIMENSION,
        sampler = optuna.samplers.NSGAIISampler(seed=seed)
    )
    study_search_space = {}
    for i in range(len(train_x[-1])):
        study_search_space[i] = optuna.distributions.UniformDistribution(problem.bounds[0][i], problem.bounds[1][i])
    for x in train_x:
        study, _, _ = add_trial_to_study(study, study_search_space, problem, x)
    return study, study_search_space
```

**Comment by nzw0301** [comment]
Could you share the minimal and executable code for the question? The shared code is incomplete; the `optimize` block, whose `n_trials` argument, would be important to know about the question. 

In addition, what do you mean by "same or worse than random suggestion"?
